### PR TITLE
Expose tantivy's TermQuery

### PR DESCRIPTION
**Pull request by mocobeta**
This may relate to #20.

This exposes Tantivy's `TermQuery` to tantivy-py.
I also noticed #53, but it seems incomplete and stale for a while.

- Add `Query.new_term_query()` method with proper type hints.
- Add a test method for TermQuery.

### CODE REVIEW

File: src/query.rs
Comment (adamreichold) [inline src/query.rs:32]: Why the separate function to implement this instead of directly here in the method?

File: src/query.rs
Comment (adamreichold) [inline src/query.rs:26]: I would suggest just `term_query`.

File: tests/tantivy_test.py
Comment (adamreichold) [inline tests/tantivy_test.py:933]: I think exposing the fundamental query types is not too useful without having combinators, i.e. we should at least add `union` and `intersection` combinators to combine queries on the Python side.

File: src/query.rs
Comment (mocobeta) [reply src/query.rs:26]: Thanks for the suggestion. `term_query` sounds more concise.

File: src/query.rs
Comment (mocobeta) [reply src/query.rs:32]: It's just for readability. I suppose we will have several factory methods like this for other query types.
But having all implementations here is also fine.

File: tests/tantivy_test.py
Comment (mocobeta) [reply tests/tantivy_test.py:933]: I guess you mean BooleanQuery by "union" and "intersection"? Exposing TermQuery itself may not help for real use cases. However, I'd like to take baby steps and not bloat this PR. 
We can support all other queries in a step-by-step manner once this is successfully merged.

Comment (adamreichold) [reply tests/tantivy_test.py:933]: Yes, they would be implemented via `BooleanQuery`, but mirror its API, there would most likely be `union` and `intersection` methods on `Query`.

Fair enough on doing this in a follow-up PR.

Comment (mocobeta) [reply tests/tantivy_test.py:933]: Sure, we should be able to have some fluent APIs to construct a bool query. It'll be a future work; anyway, thanks for your suggestion!

File: src/query.rs
Comment (mocobeta) [reply src/query.rs]: Renamed in https://github.com/quickwit-oss/tantivy-py/pull/175/commits/e02b47986fc9c6f1de836ec49a9922a83e6ffd85.

File: src/query.rs
Comment (mocobeta) [reply src/query.rs]: I moved the implementation to `term_query()` instead of using a utility function. https://github.com/quickwit-oss/tantivy-py/pull/175/commits/68bc4668e5f86cdcb47bdcfcac31c2812bd1d50f

**Review by adamreichold — APPROVED** [review]

**Comment by mocobeta** [comment]
Hi,
This has received one approval (thank you @adamreichold for reviewing!)
I don't have permission to merge this into the main branch, could anyone take care of it?

**Comment by mocobeta** [comment]
Thanks @wallies !

**Pull request merged**
